### DISCUSSION TITLE
How to disable kitty setting the same background color for all apps

### DISCUSSION THREAD

**Discussion by DUOLabs333** [kovidgoyal/kitty#6550] [General]
For the terminal editor `micro`, the background color in every terminal except kitty >0.25 is grey. However, in kitty 0.28, the background is black. I'm assuming some behavior changed in kitty. Is there some way to disable it, as it makes it hard to read some low-contrast text.

**Comment by kovidgoyal** [comment]
Terminals dont set colors, applications do. I suggest reporting the
issue to micro. Most likely it will be micro failing to detect that
kitty supports truecolors. Make sure your TERM is correct.

On Mon, Aug 14, 2023 at 10:10:25AM -0700, DUO Labs wrote:
> For the terminal editor `micro`, the background color in every terminal except kitty >0.25 is grey. However, in kitty 0.28, the background is black. I'm assuming some behavior changed in kitty. Is there some way to disable it, as it makes it hard to read some low-contrast text.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6550
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by DUOLabs333** [reply]
I know it's not a problem with micro, as if I ssh into my machine from kitty 0.25.1, the background is grey as expected. This problem also occurs with `git-delta` highlight colors (specifically red and green) --- they're much lighter than they're supposed to.

**Reply by kovidgoyal** [reply]
I know its not a problem with kitty, since kitty does not set colors.

**Comment by DUOLabs333** [comment]
Ok, I figured out what the problem is. Apparently, between two updates, kitty seems to have dropped support / started to use some unsupported feature for my OpenGL version (3.1 pretending to be 3.3). This causes the background color to not show up. `LIBGL_ALWAYS_SOFTWARE` makes it work as expected (and explains why other terminals didn't have this problem --- they didn't use the GPU), though this obviously not ideal.

I would downgrade, but the working version uses python 3.9 specifically, not python3, so initialization fails.

**Comment by DUOLabs333** [comment]
As a follow-up (since I now switched back to kitty, as wezterm has gotten too laggy to be useful for quick switching),  @kovidgoyal is there a way to tell kitty to use the software renderer? While it does use more CPU, that is a tradeoff I'm willing to make.

**Reply by kovidgoyal** [reply]
Google opengl software renderer environment variable.
